### PR TITLE
Add indexes to comments

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)
- [x] Optimization

## Description
Following https://github.com/forem/forem/pull/13401 and https://github.com/forem/forem/pull/13469 the performance on the comments query still isn't where we want it. This PR adds more indexes to the `comments` table to improve performance.

https://github.com/forem/forem/blob/3ec91dd9368357bfd523305dfd94c7d0c5977df3/app/services/search/postgres/comment.rb#L45-L55

## Related Tickets & Documents
https://github.com/forem/forem/pull/13401
https://github.com/forem/forem/pull/13469
https://ui.honeycomb.io/thepracticaldev/datasets/rails/result/s1uhihjmPKR/trace/5PkgsLrBkdX

## QA Instructions, Screenshots, Recordings
1. Run `rails db:migrate` and see the index is added.
2. Run `rails db:rollback STEP=1` and see the index is removed.

## Added tests?
- [x] No, and this is why: this is a DB migration - no tests required.

## [Forem core team only] How will this change be communicated?
- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?
I'll sync with SRE to monitor the results of this change.

confused_power_rangers

### CODE REVIEW

File: db/migrate/20210422171642_add_indexes_to_comments.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+class AddIndexesToComments < ActiveRecord::Migration[6.1]
+  disable_ddl_transaction!
+
+  def up
+    add_index :comments, :commentable_type, algorithm: :concurrently unless index_exists?(:comments, :commentable_type)
+    add_index :comments, :deleted, algorithm: :concurrently unless index_exists?(:comments, :deleted)
+    add_index :comments, :hidden_by_commentable_user, algorithm: :concurrently unless index_exists?(:comments, :hidden_by_commentable_user)
```
Comment (rhymes) [inline db/migrate/20210422171642_add_indexes_to_comments.rb:7]: non blocking and for future reference: Rails 6.1 has added `if_not_exists:true` for `add_index` and `if_exists: true` for `remove_index`

Comment (atsmith813) [inline db/migrate/20210422171642_add_indexes_to_comments.rb:7]: Now, you may be reading this and asking yourself

```
Why not use the new Rails 6.1 sorcery of `if_not_exists`/`if_exists`?
```

Agreed! I couldn't get it to work 🤣 🙈 . I don't think it works with `algorithm: :concurrently`
Screen Shot 2021-04-21 at 3 08 02 PM

It messes up the query syntax 😬

**Review by rhymes — APPROVED** [review]
LGTM!

File: db/migrate/20210422171642_add_indexes_to_comments.rb (hunk 1 above)
Comment (rhymes) [reply db/migrate/20210422171642_add_indexes_to_comments.rb:7]: oh gosh!

Comment (atsmith813) [reply db/migrate/20210422171642_add_indexes_to_comments.rb:7]: Haha you beat me to it 😅  https://github.com/forem/forem/pull/13476#discussion_r618605158

**Review by maestromac — APPROVED** [review]

**Review by djuber — APPROVED** [review]
merging

**Pull request merged**